src/components/FormStart/index.js

```diff
@@ -61,12 +74,17 @@ const FormStartMessage = ({form}) => {
  */
 const FormStart = ({ form, onFormStart }) => {
   const doStart = useStartSubmission();
+  const autoRedirectUrl = getLoginRedirectUrl(form);
   const outagePluginId = useDetectAuthenticationOutage();
   const authErrors = useDetectAuthErrorMessages();
   const hasAuthErrors = !!outagePluginId || !!authErrors;
 
   const onFormStartCalledRef = useRef(false);
 
+  if (!doStart && autoRedirectUrl) {
+    window.location = autoRedirectUrl;
+  }
```

Comment:
@stevenbal w/r to your 'flickering' questions.

1. This code belongs in the `useEffect` hook - doing redirects in the render phase is an anti-pattern iirc, it should only return JSX
2. We have a [loader](https://github.com/open-formulieren/open-forms-sdk/blob/main/src/components/Loader.js)/spinner component you can display - if these conditions are met for auto-redirect, in theory you should be able to just do `return <Loader />`. That should help with the flicker.

Reply:
Silvia fixed this, now it does show the loading indicator (thanks @SilviaAmAm!)